packages/legacy/src/Providers/components/AddEditProviderModal/AddEditProviderModal.tsx

```diff
@@ -178,20 +194,29 @@ export const AddEditProviderModal: React.FunctionComponent<IAddEditProviderModal
   const providerType = providerTypeField.value;
   const formValues = providerType ? forms[providerType].values : null;
   const isFormValid = providerType ? forms[providerType].isValid : false;
+  const isOpenStackFormValid = forms["openstack"].isValid && (forms["openstack"].fields.insecure.value || forms["openstack"].fields.caCertIfSecure.value);
   const isFormDirty = providerType ? forms[providerType].isDirty : false;
 
   // Combines fields of all 3 forms into one type with all properties as optional.
   // This way, we can conditionally show fields based on whether they are defined in form state
   // instead of duplicating the logic of which providers have which fields.
   const fields = providerType
     ? (forms[providerType].fields as Partial<
-        typeof forms.vsphere.fields & typeof forms.ovirt.fields & typeof forms.openshift.fields
+        typeof forms.vsphere.fields & typeof forms.ovirt.fields & typeof forms.openstack.fields & typeof forms.openshift.fields
       >)
     : null;
 
   const usernamePlaceholder = {
     vsphere: 'Example, administrator@vsphere.local',
     ovirt: 'Example, admin@internal',
+    openstack: 'Example, admin',
+    openshift: undefined,
```

Comment:
```suggestion
    openshift: 'Example, admin',
```
an example from the docs
https://docs.openstack.org/keystone/latest/install/keystone-install-rdo.html#configure-the-apache-http-server

Reply:
No username is set/displayed for the openShift provider. We use a token instead.